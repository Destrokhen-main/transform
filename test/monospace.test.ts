import transform from '../src/transform';
import monospace from '../src/transform/plugins/monospace';

const html = (text: string) => {
    const {
        result: {html},
    } = transform(text, {
        plugins: [monospace],
    });
    return html;
};

describe('Monospace plugin', () => {
    it('should not render header', () => {
        expect(html('## Header')).toMatchSnapshot();
    });
    it('should render simple block', () => {
        expect(html('##text##')).toMatchSnapshot();
    });
    it('should render simple block with text before', () => {
        expect(html('test ##text##')).toMatchSnapshot();
    });
    it('should render simple block with text after', () => {
        expect(html('##text## test')).toMatchSnapshot();
    });
    it('should render simple block between texts', () => {
        expect(html('test ##text## test')).toMatchSnapshot();
    });
    it('should not render if open token incorrect', () => {
        expect(html('#asd##')).toMatchSnapshot();
    });
    it('should not render if space after open token', () => {
        expect(html('## test##')).toMatchSnapshot();
    });
    it('should not render if space before close token', () => {
        expect(html('##test ##')).toMatchSnapshot();
    });
    it('should not render opened token', () => {
        expect(html('test ##test')).toMatchSnapshot();
    });
    it('should render with open token inside', () => {
        expect(html('test1 ##test2 ##test3##')).toMatchSnapshot();
    });
    it('should render inline markup', () => {
        expect(html('##test **bold**##')).toMatchSnapshot();
    });
    it('should render siblings monospace', () => {
        expect(html('##test1## ##test2##')).toMatchSnapshot();
    });
    it('should not render with empty content', () => {
        expect(html('test #### test')).toMatchSnapshot();
    });
    it('should not render last open token', () => {
        expect(html('##test1 **bold**## test2 ##test3')).toMatchSnapshot();
    });
    it('should render content with #', () => {
        expect(html('###test##')).toMatchSnapshot();
    });
});
