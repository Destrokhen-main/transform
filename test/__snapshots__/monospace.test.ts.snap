// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monospace plugin should not render header 1`] = `
<h2>
  Header
</h2>
`;

exports[`Monospace plugin should not render if open token incorrect 1`] = `
<p>
  #asd##
</p>
`;

exports[`Monospace plugin should not render if space after open token 1`] = `
<h2>
  test##
</h2>
`;

exports[`Monospace plugin should not render if space before close token 1`] = `
<p>
  ##test ##
</p>
`;

exports[`Monospace plugin should not render last open token 1`] = `
<p>
  <samp>
    test1
    <strong>
      bold
    </strong>
  </samp>
  test2 ##test3
</p>
`;

exports[`Monospace plugin should not render opened token 1`] = `
<p>
  test ##test
</p>
`;

exports[`Monospace plugin should not render with empty content 1`] = `
<p>
  test #### test
</p>
`;

exports[`Monospace plugin should render content with # 1`] = `
<p>
  #
  <samp>
    test
  </samp>
</p>
`;

exports[`Monospace plugin should render inline markup 1`] = `
<p>
  <samp>
    test
    <strong>
      bold
    </strong>
  </samp>
</p>
`;

exports[`Monospace plugin should render siblings monospace 1`] = `
<p>
  <samp>
    test1
  </samp>
  <samp>
    test2
  </samp>
</p>
`;

exports[`Monospace plugin should render simple block 1`] = `
<p>
  <samp>
    text
  </samp>
</p>
`;

exports[`Monospace plugin should render simple block between texts 1`] = `
<p>
  test
  <samp>
    text
  </samp>
  test
</p>
`;

exports[`Monospace plugin should render simple block with text after 1`] = `
<p>
  <samp>
    text
  </samp>
  test
</p>
`;

exports[`Monospace plugin should render simple block with text before 1`] = `
<p>
  test
  <samp>
    text
  </samp>
</p>
`;

exports[`Monospace plugin should render with open token inside 1`] = `
<p>
  test1 ##test2
  <samp>
    test3
  </samp>
</p>
`;
